app/models/events/pull_request_comment.rb

```diff
@@ -43,5 +43,13 @@ class PullRequestComment < ApplicationRecord
 
     validates :state, inclusion: { in: states.keys }
     validates :github_id, :body, :opened_at, presence: true
+
+    after_create :build_review_or_comment_turnaround
```

Comment:
I'm not a fan of model callbacks for multiple reasons, there are many blog posts written about their cons, here is one example https://engineering.gusto.com/the-rails-callbacks-best-practices-used-at-gusto/

>Conclusion
With great power comes great responsibility, and Rails callbacks are no exception. They are expressive tools but also come with drawbacks.
>
>We are generally moving away from using callbacks and toward using rich service objects in our applications, but we still have many callbacks that exist. We still have a lot left to do, but we’ve found that the code that follows these guidelines requires less maintenance.

Here are a couple more links:
- https://blog.arkency.com/2016/05/domain-events-over-active-record-callbacks/ 
- https://twitter.com/_solnic_/status/974214728066260993

In a few words, we are polluting a domain object with complex creational logic (this is a bit advanced but I highly recommend it https://www.infoq.com/minibooks/domain-driven-design-quickly/)

Wouldn't it be possible to have this logic in another place (e.g. a service where the PullRequest is created)

Reply:
given that we are already using callbacks in the project, I'm good leave this as it is, but keep it in mind for future PRs that could be a good idea to move away from model callbacks

Reply:
Oh! Interesting information, Thank you very much. I will of course keep it in mind for future developments.